extension/json/json_functions/json_array_rotate.cpp

```diff
@@ -0,0 +1,64 @@
+#include "json_executors.hpp"
+
+namespace duckdb {
+
+//! Rotate String or JSON value to an array
+template <class AMOUNT_TYPE>
+yyjson_mut_val *ArrayRotate(yyjson_mut_val *arr, AMOUNT_TYPE amount, yyjson_alc *alc, Vector &result) {
+	if (!yyjson_mut_is_arr(arr)) {
+		throw InvalidInputException("JSON input not an JSON Array");
+	}
+
+	size_t array_size = yyjson_mut_arr_size(arr);
+	if (amount == 0 || array_size == 0) {
+		return arr;
+	}
+
+	size_t shift_amount = 0;
+
+	if (amount < 0) {
```

Comment:
You want to bounds-check this and use `(Unsafe)NumericCast<AMOUNT_TYPE, size_t>` instead. You could maybe make use of `TryAddOperator` and `TrySubtractOperator` potentially, or just catch the internal exception thrown when NumericCast fails. `sizeof(int64_t) != sizeof(size_t)` on e.g. WASM